### ISSUE TITLE
distfiles server for guru-overlay (or for arbitrary overlay)

### ISSUE THREAD

**Issue by fpemud** [gentoo/guru#41]
Gentoo distfiles server are way much faster than the original source.
But for packages in overlay, currently, fetching from the original source is the only way.
Is there any plan for guru-overlay having a distfiles server? Or is there any software that be used as distfiles server for arbitrary overlay?
This question has been discussed in Gentoo forum (https://forums.gentoo.org/viewtopic-t-1122071-start-0.html) but there's no clear solution.

**Comment by AndrewAmmerlaan** [comment]
AFAIK, there are no plans for this at the moment. And AFAIK it is not possible to define an url to a distfiles server in the overlay itself, it would have to be done via `GENTOO_MIRRORS` which is far from ideal.

Perhaps something like this could be an enhancement to the `repos.conf` structure, e.g. a per repository disfiles server in the relevant repos.conf section.

**Comment by jian-lin** [comment]
@fpemud `emirrordist`  is what you are looking for. It is included in `sys-apps/portage`. I learn about it from `#gentoo` channel on freenode.

**Comment by fpemud** [comment]
Thanks. I think it's possible to build an overlay-distfile-server using this tool.
But I'm still curious how distfiles.gentoo.org fetches all the files? Is it using emirrordist?

**Comment by Alessandro-Barbieri** [comment]
I don't think this is the right place to discuss. It's better to open a ticket on bugs.gentoo.org and ask there

**Comment by Alessandro-Barbieri** [comment]
see also https://bugs.gentoo.org/790509
